src/main/java/PersonalBudget/business/income/domain/service/IncomeService.java

```diff
@@ -1,39 +1,60 @@
 import PersonalBudget.business.income.domain.mapper.IncomeMapper;
+import PersonalBudget.business.income.domain.model.DefaultCategory;
+import PersonalBudget.business.income.domain.model.IncomeCategoryEntity;
 import PersonalBudget.business.income.domain.model.IncomeEntity;
 import PersonalBudget.business.income.domain.repository.IncomeCategoryRepository;
 import PersonalBudget.business.income.domain.repository.IncomeRepository;
+import PersonalBudget.business.income.domain.service.exception.CategoryIdNotFoundException;
 import PersonalBudget.business.income.dto.IncomeDTO;
-import PersonalBudget.business.user.domain.repository.UserRepository;
+import PersonalBudget.business.user.domain.UserFacade;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import java.util.Arrays;
 import java.util.List;
 
+
+@RequiredArgsConstructor
 @Service
-public class IncomeService extends BaseService{
+public class IncomeService {
 
     private final IncomeCategoryRepository incomeCategoryRepository;
     private final IncomeRepository incomeRepository;
     private final IncomeMapper incomeMapper;
-
-    public IncomeService(UserRepository userRepository, IncomeCategoryRepository incomeCategoryRepository,
-                         IncomeRepository incomeRepository, IncomeMapper incomeMapper) {
-        super(userRepository);
-        this.incomeRepository = incomeRepository;
-        this.incomeCategoryRepository = incomeCategoryRepository;
-        this.incomeMapper = incomeMapper;
-    }
+    private final UserFacade userFacade;
 
     public List<String> getCategoriesAssignedToUser() {
-        return incomeCategoryRepository.getAllIncomeCategories(getCurrentLoggedInUserId());
+        Long loggedInUserId = userFacade.fetchLoggedInUserId();
+        return incomeCategoryRepository.findAllIncomeCategoryName(loggedInUserId);
     }
 
     public void addIncome(IncomeDTO incomeDTO) {
-        Long userId = getCurrentLoggedInUserId();
-        Long userIncomeCategory = incomeCategoryRepository.getIdOfParticularCategory(userId, incomeDTO.category());
+        Long userId = userFacade.fetchLoggedInUserId();
+        Long userIncomeCategory = incomeCategoryRepository.findCategoryIdByUserIdAndCategoryName(userId, incomeDTO.category()).orElseThrow(() ->
+                new CategoryIdNotFoundException("Category id not found"));
         IncomeEntity incomeEntity = incomeMapper.mapIncomeDTOToIncomeEntity(incomeDTO, userId, userIncomeCategory);
         incomeRepository.save(incomeEntity);
     }
+
+    public List<IncomeCategoryEntity> buildDefaultCategories(Long userId) {
+        return Arrays.stream(DefaultCategory.values()).map(defaultCategory -> IncomeCategoryEntity.builder()
+                .userId(userId)
+                .name(defaultCategory.toString().toLowerCase())
+                .build())
+                .toList();
+
+        /*return DefaultCategory.stream().map(defaultCategory -> IncomeCategoryEntity.builder()
```

Comment:
Why not this implementation? The builder should be indented to improve readability ;)